src/BlockTypes/AbstractProductGrid.php

```diff
@@ -68,57 +126,79 @@ protected function parse_attributes( $attributes ) {
 			),
 		);
 
-		$attributes = wp_parse_args( $attributes, $defaults );
-
-		if ( ! empty( $attributes['rows'] ) && ! empty( $attributes['columns'] ) ) {
-			$attributes['limit'] = intval( $attributes['columns'] ) * intval( $attributes['rows'] );
-		} else {
-			$attributes['limit'] = -1;
-		}
-
-		return $attributes;
+		return wp_parse_args( $attributes, $defaults );
 	}
 
 	/**
-	 * Set args specific to this block
-	 *
-	 * @param array $query_args Query args.
-	 */
-	abstract protected function set_block_query_args( &$query_args );
-
-	/**
-	 * Set orderby/order query args.
+	 * Parse query args.
 	 *
-	 * @param array $query_args Query args.
+	 * @return array
 	 */
-	protected function set_ordering_query_args( &$query_args ) {
-		$orderby = '';
-		$order   = '';
+	protected function parse_query_args() {
+		$query_args = array(
+			'post_type'           => 'product',
+			'post_status'         => 'publish',
+			'fields'              => 'ids',
+			'ignore_sticky_posts' => true,
+			'no_found_rows'       => false,
+			'orderby'             => '',
+			'order'               => '',
+		);
 
 		if ( isset( $this->attributes['orderby'] ) ) {
 			if ( 'price_desc' === $this->attributes['orderby'] ) {
-				$orderby = 'price';
-				$order   = 'DESC';
+				$query_args['orderby'] = 'price';
+				$query_args['order']   = 'DESC';
 			} elseif ( 'price_asc' === $this->attributes['orderby'] ) {
-				$orderby = 'price';
-				$order   = 'ASC';
+				$query_args['orderby'] = 'price';
+				$query_args['order']   = 'ASC';
+			} elseif ( 'date' === $this->attributes['orderby'] ) {
+				$query_args['orderby'] = 'date';
+				$query_args['order']   = 'DESC';
 			} else {
-				$orderby = $this->attributes['orderby'];
+				$query_args['orderby'] = $this->attributes['orderby'];
 			}
 		}
 
-		// This handles orderby queries and hooks in custom orderby functions.
-		$query_args = array_merge( $query_args, WC()->query->get_catalog_ordering_args( $orderby, $order ) );
+		if ( ! empty( $this->attributes['rows'] ) ) {
+			$this->attributes['limit'] = intval( $this->attributes['columns'] ) * intval( $this->attributes['rows'] );
+		}
+
+		$query_args['posts_per_page'] = intval( $this->attributes['limit'] );
+		$query_args['meta_query']     = WC()->query->get_meta_query(); // phpcs:ignore WordPress.DB.SlowDBQuery
+		$query_args['tax_query']      = array(); // phpcs:ignore WordPress.DB.SlowDBQuery
+
+		$this->set_block_query_args( $query_args );
+
+		$ordering_args         = WC()->query->get_catalog_ordering_args( $query_args['orderby'], $query_args['order'] );
+		$query_args['orderby'] = $ordering_args['orderby'];
+		$query_args['order']   = $ordering_args['order'];
+		if ( $ordering_args['meta_key'] ) {
+			$query_args['meta_key'] = $ordering_args['meta_key']; // phpcs:ignore WordPress.DB.SlowDBQuery
+		}
```

Comment:
From the previous implementation, I liked it had a `set_ordering_query_args()` method that was keeping `get_products_query_args()` (now `parse_query_args()`) pretty small and easy to read.

Do you think it would be possible to extract order query code (from line 148 to line 178, I guess) to its own method?

Reply:
Yes, let's try to do that and simplify this method in another PR and get this one merged; that will unblock other block work in the meantime :)